### PR TITLE
Mark CSI StorageClass as the default one

### PR DESCRIPTION
**Pull request by RomanBednar**
Bump library-go and use StorageClass controller instead of static resource controller.

cc @openshift/storage

### CODE REVIEW

**Comment by jsafrane** [comment]
IMO this is not related to https://bugzilla.redhat.com/show_bug.cgi?id=2066665, but to CSI migration.
/retitle Mark CSI StorageClass as the default one

**Comment by jsafrane** [comment]
@RomanBednar Please also remove the in-tree storage class creation from [cluster-storage-operator](https://github.com/openshift/cluster-storage-operator/tree/master/assets/storageclasses) (+ [here](https://github.com/openshift/cluster-storage-operator/blob/master/pkg/operator/defaultstorageclass/controller.go#L187))

File: assets/storageclass.yaml [hunk 1]
```diff
@@ -2,6 +2,8 @@ apiVersion: storage.k8s.io/v1
 kind: StorageClass
 metadata:
   name: managed-csi
+  annotations:
+    storageclass.kubernetes.io/is-default-class: "true"
```
Comment (bertinatto) [inline assets/storageclass.yaml:6]: When the cluster is updated, and a default SC is already set, won't this leave the cluster with 2 SCs?

Comment (bertinatto) [reply assets/storageclass.yaml:6]: Also, if we do this, should we change the Stack Hub SC as well?

Comment (RomanBednar) [reply assets/storageclass.yaml:6]: > When the cluster is updated, and a default SC is already set, won't this leave the cluster with 2 SCs?

Depends on which SC is set as default, if there is one with a different name set as default, we create ours as non default. If there already is one with the same name as the one we try to apply the new controller copies it's "is-default-class" annotation before apply is called. So yes, there can be two different SCs, but only one of them will be default after update (unless user overrides it) - which is what we wanted I believe. Is that what you meant?

> Also, if we do this, should we change the Stack Hub SC as well?

I don't know.... @jsafrane ?

Comment (jsafrane) [reply assets/storageclass.yaml:6]: The new SC controller *should* mark this (`managed-csi`) as not default when there is already another default SC. Or when the SC already exists and is not default.

**Comment by RomanBednar** [comment]
> @RomanBednar Please also remove the in-tree storage class creation from [cluster-storage-operator](https://github.com/openshift/cluster-storage-operator/tree/master/assets/storageclasses) (+ [here](https://github.com/openshift/cluster-storage-operator/blob/master/pkg/operator/defaultstorageclass/controller.go#L187))

Done: https://github.com/openshift/cluster-storage-operator/pull/272

**Comment by bertinatto** [comment]
/override  ci/prow/e2e-azurestack-csi 
Azure Stack Hub job is broken and will be fixed once the timeouts are bumped (via rebase).

**Pull request merged**